resharper/resharper-unity/test/src/CSharp/UnityVersionTest.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Linq;
+using NUnit.Framework;
+
+namespace JetBrains.ReSharper.Plugins.Unity.Tests
+{
+    [TestFixture]
+    public class UnityVersionTest
+    {
+        [Test]
+        public void VersionConversionTest()
+        {
+            var marketingVersion = "2018.2.13p1";
+            var realVersion = Version.Parse("2018.2.13.1121");
+            Assert.AreEqual(realVersion, Unity.UnityVersion.Parse(marketingVersion));
+            Assert.AreEqual(marketingVersion, Unity.UnityVersion.VersionToString(realVersion));
+        }
+        
+        [Test]
+        public void VersionToStringTest()
+        {
+            var marketingVersion = "2018.2.13";
+            var realVersion = Version.Parse("2018.2.13.0");
+            
+            Assert.AreEqual(marketingVersion, Unity.UnityVersion.VersionToString(realVersion));
+        }
+
+        [Test]
+        public void SortTest()
+        {
+            var mVersions = new[] {"2018.2.13p1", "2018.2.13a20", "2018.2.13f20", "2018.2.13b20", "2018.2.13p20"};
```

Comment:
Add in some extra values with different ranges, e.g. `f1` and `b10` that would break with the existing rules